tools/src/bin/fap-lld/main.rs

```diff
@@ -0,0 +1,56 @@
+use std::{
+    env, io,
+    path::PathBuf,
+    process::{self, Command},
+};
+
+use elf::ParseError;
+use which::which;
+
+mod fastrel;
+
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Parse(ParseError),
+    NoSymbolTable,
+    NoSectionHeaders,
+    ObjcopyFailed,
+}
+
+impl From<io::Error> for Error {
+    fn from(e: io::Error) -> Self {
+        Error::Io(e)
+    }
+}
+
+impl From<ParseError> for Error {
+    fn from(e: ParseError) -> Self {
+        Error::Parse(e)
+    }
+}
+
+fn main() -> Result<(), Error> {
+    // Run the real linker with the given arguments.
+    let res = Command::new("rust-lld")
+        .args(env::args_os().skip(1))
+        .status()?;
+    if !res.success() {
+        process::exit(res.code().unwrap_or(-1));
```

Comment:
Similar to the Python code, [`ExitStatus::code`](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code) will give `Some(0)` on success, so we can remote the [`ExitStatus::success`](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.success) conditional.

Reply:
Here we cannot, because if this succeeds, we want to continue on and perform the post-linker optimizations.